src/stream/content_internal.ml

```diff
@@ -172,15 +172,17 @@ module VideoSpecs = struct
       | _ -> None
 
   let merge p p' =
+    let f = Option.map Lazy.force in
+    let g = Option.map Lazy.from_val in
     {
-      width = merge_param ~name:"width" (p.width, p'.width);
-      height = merge_param ~name:"height" (p.height, p'.height);
+      width = g (merge_param ~name:"width" (f p.width, f p'.width));
+      height = g (merge_param ~name:"height" (f p.height, f p'.height));
     }
 
   let compatible p p' =
     let compare = function
       | None, None -> true
-      | Some _, None | None, Some _ -> false
+      | Some _, None | None, Some _ -> true
```

Comment:
In fact "correct" or not depends on the semantics you give to `None`: it could be_some_ fixed format or _any_ format. My commit changes from the first to the second. I don't think we used the difference in any relevant way, and the second one was the most simple way of handling decoders: a decoder can decode at _any_ required format.

Reply:
Sounds. To me, as long as we have a test for it, we make sure to be consistent with our implementation, which is what matters most.